fsi/init.go

```diff
@@ -131,7 +131,7 @@ func (fsi *FSI) InitDataset(p InitParams) (name string, err error) {
 		}
 	}
 
-	if err = fsi.repo.Logbook().WriteDatasetInit(context.TODO(), name); err != nil {
+	if err = fsi.repo.Logbook().WriteDatasetInit(context.TODO(), ref.Name); err != nil {
```

Comment:
@dustmop case-and-point when dealing with strings. This used to write a `name` string, which was passing in `peername/name` here, and causing silent creation of a _second_ logbook reference for `peername/name`. Given that logbook doesn't do any validation in relation to naming, we should either add some sort of logbook-level validation, or find some way to ensure the data structures passed to logbook are uniform and ideally always valid.